ImageD11/sinograms/dataset.py

```diff
@@ -85,7 +96,59 @@ def __repr__(self):
             r.append( '# scans %d from %s to %s'%(
                 len(self.scans), self.scans[0], self.scans[-1] ))
         return  "\n".join(r)
+    
+    def guess_detector(self):
+        '''Guess which detector we are using from the masterfile'''
+        
+        # open the masterfile
+        hname = self.masterfile
+        detectors_seen = []
+        scan = "1.1"
 
+        with h5py.File( hname, 'r' ) as hin:
+            # go through the first scan, and see what detectors we can see
+            for measurement in list(hin[scan]['measurement']):
```

Comment:
for item in hin[scan]['measurement']:
    if "interpretation" in item.attrs and item.attrs[ "interpretation" ] == "image":
         ....